api-registration/server/index.js

```diff
@@ -0,0 +1,64 @@
+require('dotenv/config');
+const pg = require('pg');
+const argon2 = require('argon2'); // eslint-disable-line
+const express = require('express');
+const ClientError = require('./client-error');
+const errorMiddleware = require('./error-middleware');
+
+const db = new pg.Pool({ // eslint-disable-line
+  connectionString: process.env.DATABASE_URL,
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
+
+const app = express();
+
+const jsonMiddleware = express.json();
+
+app.use(jsonMiddleware);
+
+app.post('/api/auth/sign-up', (req, res, next) => {
+  const { username, password } = req.body;
+  if (!username || !password) {
+    throw new ClientError(400, 'username and password are required fields');
+  }
+  /* your code starts here */
+
+  argon2.hash(password)
+    .then(result => {
+      const sql = `
+    INSERT INTO "users" ("username", "hashedPassword")
+    VALUES ($1, $2)
+    RETURNING *;
+    `;
+      const params = [username, result];
+      db.query(sql, params)
+        .then(result => {
+          const newPassword = result.rows;
+          res.status(201).json(newPassword);
+        })
```

Comment:
The client should receive a JSON object instead of an array. Also, they shouldn't receive the hashedPassword in the response.